### PR TITLE
[ecosystem] Add settings pages.

### PR DESCRIPTION
**Pull request by zacharydenton**
### Description

Adds settings pages for username, email, and oauth connections. These aren't linked from anywhere yet. We will need to follow up with logic to merge accounts on the oauth settings page.

https://user-images.githubusercontent.com/310773/174398298-ca42c03b-c02f-4b51-ab85-8f279b46f9f2.mp4

### Test Plan
I wrote automated tests.

### CODE REVIEW

File: ecosystem/platform/server/app/models/user.rb [hunk 1]
```diff
@@ -116,8 +117,13 @@ def email_required?
...
-  # This is to allow username instead of email login in devise (for aptos admins)
-  def will_save_change_to_email?
-    false
+  # Use mailchimp instead of the default devise confirmation email.
+  def send_confirmation_instructions
+    generate_confirmation_token! unless @raw_confirmation_token
+
+    url_options = Rails.application.config.action_mailer.default_url_options
+    url = Rails.application.routes.url_helpers.user_confirmation_url(**url_options,
+                                                                     confirmation_token: @raw_confirmation_token)
```
Comment (CapCap) [inline ecosystem/platform/server/app/models/user.rb:126]: is `raw_confirmation_token` something Devise adds?

File: ecosystem/platform/server/app/models/user.rb [hunk 2]
```diff
@@ -11,7 +11,8 @@ class User < ApplicationRecord
          :omniauthable, omniauth_providers: %i[discord github google],
                         authentication_keys: [:username]
 
-  validates :username, uniqueness: { case_sensitive: false }, allow_nil: true
+  validates :username, uniqueness: { case_sensitive: false }, length: { maximum: 20 },
```
Comment (CapCap) [inline ecosystem/platform/server/app/models/user.rb:14]: can we validate minimum length of 3 as well? (discourse thing)

File: ecosystem/platform/server/app/controllers/settings_controller.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    end
+  end
+
+  def connections
+    store_location_for(current_user, request.path)
+    @authorizations = @user.authorizations.to_h do |authorization|
+      [authorization.provider, authorization]
+    end
+
+    return unless request.method_symbol == :post
```
Comment (CapCap) [inline ecosystem/platform/server/app/controllers/settings_controller.rb:34]: this is enforceable via rails routes, right?

File: ecosystem/platform/server/app/models/user.rb (hunk 1 above)
Comment (zacharydenton) [reply ecosystem/platform/server/app/models/user.rb:126]: Yes, the first half of the method is copied from the original implementation in Devise.

File: ecosystem/platform/server/app/models/user.rb (hunk 2 above)
Comment (zacharydenton) [reply ecosystem/platform/server/app/models/user.rb]: Done.

File: ecosystem/platform/server/app/controllers/settings_controller.rb (hunk 3 above)
Comment (zacharydenton) [reply ecosystem/platform/server/app/controllers/settings_controller.rb]: Yeah it is probably a better idea to have separate methods for showing & updating the settings. Updated.

**Review by moonclavedev — APPROVED** [review]

File: ecosystem/platform/server/app/views/settings/connections.html.erb [hunk 4]
```diff
@@ -0,0 +1,31 @@
+<% content_for(:page_title, 'Connection Settings') %>
+<h3 class="text-4xl mb-8 font-display font-light">Connections</h3>
+<% User.omniauth_providers.each do |provider| %>
+  <div class="mb-8 flex justify-between">
+    <div class="flex gap-4 items-center">
+      <%= render IconComponent.new(provider, size: :large) %>
+      <div>
+        <h4 class="text-xl font-display font-light"><%= provider.to_s.titleize %></h4>
+        <% if @authorizations[provider.to_s] %>
+          <p><%= @authorizations[provider.to_s].email || @authorizations[provider.to_s].username %></p>
```
Comment (CapCap) [inline ecosystem/platform/server/app/views/settings/connections.html.erb:10]: Can we make a method on authorizations to specify the display value? i.e email for gmail, username for discord + github, etc

Comment (zacharydenton) [reply ecosystem/platform/server/app/views/settings/connections.html.erb:10]: Done.

**Pull request closed without merging**